### ISSUE TITLE
[crash/fuzzing] "index out of bounds" in wain validate

### ISSUE THREAD

**Issue by pventuzelo** [rhysd/wain#39]
Hey,

My fuzzer just triggered this panic `index out of bounds` error.

Crash:
``` sh
$ ./target/debug/wain index_oob_wain.wasm 
thread 'main' panicked at 'index out of bounds: the len is 4 but the index is 93', /home/scop/Documents/wain/wain-validate/src/insn.rs:391:28
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Download:
[index_oob_wain.zip](https://github.com/rhysd/wain/files/4887064/index_oob_wain.zip)

Bug is happening here:
https://github.com/rhysd/wain/blob/27f9ef475ab29adc267b51225ab48288cc498ea3/wain-validate/src/insn.rs#L390-L392

**Comment by rhysd** [comment]
Thank you for catching this. I'll look into.

**Comment by rhysd** [comment]
This issue was fixed at 08527f5. Now it raises validation error as follows:

```
Error on validation: type index 93 out of bounds 0 <= idx < 4. error while validating callee at call instruction.  caused at byte offset 562

 ... 34 36 37 34 34 30 37 33 37 30 39 35 35 31 36 31 35 01 2b 26 3c 30 81 90 9b
     ^
     starts from here
```
